lucene/test-framework/src/java/org/apache/lucene/analysis/TokenStreamUtil.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.analysis;
+
+import java.io.PrintStream;
+
+public final class TokenStreamUtil {
+
+  private TokenStreamUtil() {
+  }
+
+  /**
+   * Prints details about consumed tokens stored in any ValidatingTokenFilters in the input chain
+   * @param in
+   * @param out
+   */
+  public static void dumpValidatingTokenFilters(TokenStream in, PrintStream out) {
```

Comment:
I think this method can just be a private method on ValidatingTokenFilter?  No need for the extra class

Reply:
Yes - I started from a broader project including the ability to inject arbitrary filters into any token stream. I was thinking of `ToDotTokenFilter`, but simplified it along the way since I didn't need that for debugging here. I'll move into VTF and create this later as needed.